src/main/java/io/frictionlessdata/tableschema/field/YearField.java

```diff
@@ -30,7 +32,10 @@ public YearField(String name, String format, String title, String description,
 
     @Override
     public Year parseValue(String value, String format, Map<String, Object> options) throws InvalidCastException, ConstraintsException {
-        Pattern pattern = Pattern.compile(REGEX_YEAR);
+        if(value.length()>4){
```

Comment:
I believe this check is not needed, as the `pattern.matcher` will throw anyways.

I whipped up a small test:
     @Test
     @DisplayName("format year")
     void formatNumberField() {
         YearField field = new YearField("test");
         Year val = field.parseValue("12345", null, null);
     }

And yes, it throws. Is there something I am missing or is this check just a precaution?

Reply:
It was intended as a precondition to avoid calling the `pattern.matcher` which comes with some computational overhead, but I'll take it away since the gains are negligible against how it affects readability.